spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaProducerFactory.java

```diff
@@ -867,7 +867,7 @@ protected Producer<K, V> createTransactionalProducer(String txIdPrefix) {
 	private boolean expire(CloseSafeProducer<K, V> producer) {
 		boolean expired = this.maxAge > 0 && System.currentTimeMillis() - producer.created > this.maxAge;
 		if (expired) {
-			closeTransactionProducer(producer, this.physicalCloseTimeout, this.listeners);
+			producer.closeDelegate(this.physicalCloseTimeout, this.listeners);
```

Comment:
At https://github.com/spring-projects/spring-kafka/blob/2306ac5cbd47702f9a492541186944682edada9c/spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaProducerFactory.java#L846-L853

`expire()` is called by A, in this case producer need close and return suffix

Reply:
Sure, but `expire()` will call `closeDelegate`, which will, in turn, call `remover`, and in the case of a transactional producer, it is the `cacheReturner`.

Reply:
Rework this behaviour. Look at [here](https://github.com/spring-projects/spring-kafka/pull/2930#discussion_r1428924996).